packages/scheduler/src/WrappedTarget.ts

```diff
@@ -146,33 +148,27 @@ export class WrappedTarget implements TargetRun {
       const cacheEnabled = target.cache && shouldCache && result.hash;
       // Save output if cache is enabled & cache is hit
       if (!result.skipped && cacheEnabled) {
-        const outputLocation = getLageOutputCacheLocation(this.options.root, result.hash);
+        const outputLocation = getLageOutputCacheLocation(root, result.hash);
         const outputPath = path.dirname(outputLocation);
         await mkdir(outputPath, { recursive: true });
 
         const output = `${result.stdoutBuffer}\n${result.stderrBuffer}`;
 
         await writeFile(outputLocation, output);
 
-        this.options.logger.verbose(`>> Saved cache - ${result.hash}`, { target });
+        logger.verbose(`>> Saved cache - ${result.hash}`, { target });
       }
 
       if (result.skipped) {
         const { hash } = result;
 
-        const cachedOutputFile = getLageOutputCacheLocation(this.options.root, hash ?? "");
+        const cachedOutputFile = getLageOutputCacheLocation(root, hash ?? "");
 
-        if (fs.existsSync(cachedOutputFile)) {
-          const cachedOutput = fs.createReadStream(cachedOutputFile, "utf8");
-          this.options.logger.verbose(">> Replaying cached output", { target });
-          this.options.logger.stream(LogLevel.verbose, cachedOutput, { target });
-
-          return await new Promise<void>((resolve) => {
-            cachedOutput.on("close", () => {
-              this.onSkipped(hash);
-              resolve();
-            });
-          });
+        const shouldShowCachedOutput = fs.existsSync(cachedOutputFile);
+        if (shouldShowCachedOutput) {
+          const cachedOutput = fs.readFileSync(cachedOutputFile, "utf8");
+          logger.verbose(">> Replaying cached output", { target });
+          logger.verbose(cachedOutput.trim(), { target });
```

Comment:
Should this be `logger.stream`?

Reply:
I did the stream before - and the problem was that the close of the stream fought with the idea of being able to tell when the stream was consumed completely from the logger and ended up having summary printed before all the log streams were printed... at least that's ONE source of that mess!